internal/controllers/synthetic_devices_controller.go

```diff
@@ -264,33 +271,24 @@ func (vc *SyntheticDevicesController) MintSyntheticDevice(c *fiber.Ctx) error {
 
 	rawPayload := vc.getEIP712Mint(int64(integration.TokenId), vid)
 
-	hash, _, err := signer.TypedDataAndHash(*rawPayload)
+	h, _, err := signer.TypedDataAndHash(*rawPayload)
 	if err != nil {
 		vc.log.Err(err).Msg("Error occurred creating has of payload")
 		return fiber.NewError(fiber.StatusBadRequest, "Couldn't verify signature.")
 	}
+	hash := crypto.Keccak256Hash(h)
```

Comment:
I'm worried about `h` and `hash` here. I think `TypedDataAndHash` [is already hashing](TypedDataAndHash). They don't return a `common.Hash` for whatever reason. I'd be shocked if line 282 succeeded.